app/features/user/guest-account-storage.ts

```diff
@@ -0,0 +1,48 @@
+const storageKey = 'guestAccount';
+
+type GuestAccountDetails = {
+  id: string;
+  password: string;
+};
+
+const assertGuestAccountDetails = (
+  value: unknown,
+): value is GuestAccountDetails => {
+  return (
+    value instanceof Object &&
+    'id' in value &&
+    typeof value.id === 'string' &&
+    'password' in value &&
+    typeof value.password === 'string'
+  );
+};
+
+const getGuestAccount = (): GuestAccountDetails | null => {
+  const dataString = localStorage.getItem(storageKey);
+  if (!dataString) {
+    return null;
+  }
+  const dataObject = JSON.parse(dataString);
+  if (!assertGuestAccountDetails(dataObject)) {
```

Comment:
JSON.parse also might throw an error. Is it safe to assume that dataString will always be valid JSON?

Reply:
but we could also add try/catch just in case

Reply:
added